cmd/snap-confine/snap-confine.c

```diff
@@ -386,29 +386,25 @@ int main(int argc, char **argv)
 		    " but should be. Refusing to continue to avoid"
 		    " permission escalation attacks");
 	}
-	// TODO: check for similar situation and linux capabilities.
-	if (geteuid() == 0) {
```

Comment:
I reckon in the PR description you're referring to this existing check:
```
	// This code always needs to run as root for the cgroup/udev setup.
	if (geteuid() != 0) {
```

With that, can geteuid() part of the condtion above be simplified too?

```
	if (!apparmor.is_confined && apparmor.mode != SC_AA_NOT_APPLICABLE
	    && getuid() != 0 && geteuid() == 0) {
```

Reply:
I'm not sure I follow, are you recommending to merge this with the condition above that checks if snap-confine is not confined but should be? Those are separate checks and I would not combine them simply because they produce different error messages. If I misunderstood then please clarify.

Reply:
@zyga, I think @stolowski's point is that from line 357 down you know that `geteuid() == 0` so you might as well drop that check

Reply:
Yes, this.

Reply:
Ah, I misunderstood then. Checking